torch/csrc/jit/autodiff.h

```diff
@@ -12,4 +12,21 @@ namespace torch { namespace jit {
 // (aka backward) of inputs to the first stage w.r.t. the outputs of the first stage.
 void differentiate(std::shared_ptr<Graph>& graph);
 
+using value_list = std::vector<Value*>;
+struct LiftedReverse {
+  std::shared_ptr<Graph> f;
+  std::shared_ptr<Graph> df;
+
+  // See liftLambdaReverse for a detailed overview of the structure of
+  // f's outputs and df's inputs. The gist of it is that:
+  // - f_output_intermediates describes how many temporaries were appended to outputs of f
+  //   (because they are needed by df; they should be discarded before returning f's outputs)
+  // - df_input_captures describes which outputs of f to capture and later provide
+  //   as first df_input_captures.size() inputs to df (with trailing inputs being
+  //   vjps aka grad outputs).
+  std::size_t f_output_intermediates;
+  std::vector<std::size_t> df_input_captures;
```

Comment:
aka `captured_inputs`, in the pseudocode. But I am a bit confused; don't you also need input captures? Unless you are making `f` forward those through but then that needs to be remarked upon here.